src/setup_payload/tests/TestAdditionalDataPayload.cpp

```diff
@@ -116,9 +115,9 @@ void TestGeneratingAdditionalDataPayloadWithRotatingDeviceIdAndMaxLifetimeCounte
 
     char output[kAdditionalDataPayloadLength];
     NL_TEST_ASSERT(inSuite,
-                   GenerateAdditionalDataPayload(inSuite, std::numeric_limits<uint16_t>::max(), kSerialNumber,
-                                                 strlen(kSerialNumber), additionalDataFields, output) == CHIP_NO_ERROR);
-
+                   GenerateAdditionalDataPayload(inSuite, std::numeric_limits<uint16_t>::max(),
+                                                 MutableByteSpan{ const_cast<uint8_t *>(kUniqueId), sizeof(kUniqueId) },
```

Comment:
Should not need the const_cast if the type the callee expects is fixed, here and all the other places in this file.